src/ng/q.js

```diff
@@ -361,21 +350,27 @@ function qFactory(nextTick, exceptionHandler) {
     },
 
     $$resolve: function(val) {
-      var then, fns;
+      if (this.promise.$$state.status > 0) return;
 
-      fns = callOnce(this, this.$$resolve, this.$$reject);
+      if (isPromiseLike(val)) {
```

Comment:
Previously it was `(isObject(val) || isFunction(val)) && val && isFunction(val.then)`.  Now it is `val && isFunction(val.then)`.  Is the first one more correct then the second?